tests/ci/benchmark_framework/compare_results.py

```diff
@@ -0,0 +1,63 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import pandas as pd
+import numpy as np
+
+if len(sys.argv) != 4:
+    print("Usage: compare_results.py [file1] [file2] [output filename]", file=sys.stderr)
+    sys.exit(1)
+
+file1 = sys.argv[1]
+file2 = sys.argv[2]
+
+if not (file1.endswith(".json") or file1.endswith(".csv")) and not (file2.endswith(".json") or file2.endswith(".csv")):
+    print("Provided files must either be .json files or .csv files", file=sys.stderr)
+    sys.exit(1)
+
+# read contents of file1 into a dataframe in preparation for comparison
+if file1.endswith(".json"):
+    df1 = pd.read_json(file1)
+else:
+    # this assumes we're using a csv generated by convert_json_to_csv.py
+    df1 = pd.read_csv(file1, skiprows=1, index_col=0)
+
+# read contents of file2 into a dataframe in preparation for comparison
+if file2.endswith(".json"):
+    df2 = pd.read_json(file2)
+else:
+    # this assumes we're using a csv generated by convert_json_to_csv.py
+    df2 = pd.read_csv(file2, skiprows=1, index_col=0)
+
+# only compare benchmarks that appear in both of the files
+if df1.shape[0] > df2.shape[0]:
+    df1 = df1[df1.iloc[:, 0].isin(df2.iloc[:, 0])]
+elif df2.shape[0] > df1.shape[0]:
+    df2 = df2[df2.iloc[:, 0].isin(df1.iloc[:, 0])]
+
+df1_numCalls = df1.iloc[:, 1]
+df2_numCalls = df2.iloc[:, 1]
```

Comment:
Here is an example of the expected structure of a file that this script will be run on. I plan on adding something to the script that states that it assumes that it is taking output from either the json output of the speed tool or the output of convert_json_to_csv.py.
[amd64-aws-lc-pr_bm.csv](https://github.com/awslabs/aws-lc/files/6925608/amd64-aws-lc-pr_bm.csv)

Reply:
> 2nd column of the dataframe represents the number of calls

Using index `1` is because there is another expectation/assumption on the output of benchmark script? `tests/ci/benchmark_framework/convert_json_to_csv.py` has no related mapping logics.

Reply:
There is no explicit logic for this because it is handled by the Pandas API that I am using. When the .json output of the speed tool is converted to a Pandas Dataframe, column index 0 will hold the description of the algorithm, index 1 will hold the number of calls for that benchmark, index 2 will hold the precise number of milliseconds, and index 3 will hold the number of bytes per call (if applicable). I can mark that in a comment in convert_json_to_csv.py if you'd like.

Reply:
Can you check if the JSON struct change(e.g. field 1 is moved after field 2) can change the CSV column header order? I am not sure if the Pandas API guarantees the column order by sorting the json keys (field 1 and field 2).

Reply:
Checked, changing the JSON structure resulted in the expected change in the CSV column header order. (I switched description and numCalls in the json).